File: footystats-tools-frontend/src/app/types/prediction/BetPredictionContext.ts
import { LeagueStats } from "../stats/LeagueStats";
import { MatchStats } from "../stats/MatchStats";
import TeamStats from "../stats/TeamStats";

export type BetPredictionContext = {
	match: MatchStats;
	teamStats?: TeamStats[];
	leagueStats?: LeagueStats;
	bet: Bet;
};

export enum Bet {
	OVER_ZERO_FIVE,
	OVER_ONE_FIVE,
	BTTS_YES,
	BTTS_NO,
}

File: footystats-tools-backend/src/main/java/de/footystats/tools/services/prediction/influencer/BetPredictionContext.java
package de.footystats.tools.services.prediction.influencer;

import de.footystats.tools.services.prediction.Bet;
import de.footystats.tools.services.stats.LeagueStats;
import de.footystats.tools.services.stats.MatchStats;
import de.footystats.tools.services.stats.TeamStats;

/**
 * Context for a bet prediction.
 *
 * @param match
 * @param homeTeamStats
 * @param awayTeamStats
 * @param leagueStats
 * @param bet
 */
public record BetPredictionContext(MatchStats match,
								   TeamStats homeTeamStats,
								   TeamStats awayTeamStats,
								   LeagueStats leagueStats,
								   Bet bet) {

}
